db/background_mgr_resync_dcp.go

```diff
@@ -242,6 +242,22 @@ func (r *ResyncManagerDCP) Run(ctx context.Context, options map[string]interface
 		base.InfofCtx(ctx, base.KeyAll, "[%s] resync was terminated. Docs changed: %d Docs Processed: %d", resyncLoggingID, r.DocsChanged.Value(), r.DocsProcessed.Value())
 	}
 
+	collectionsRequiringResync := make([]base.ScopeAndCollectionName, 0)
+	for _, collection := range db.CollectionByID {
+		dataStore := collection.dataStore
+		resyncRequired, err := base.InitSyncInfo(dataStore, db.DatabaseContext.Options.MetadataID)
```

Comment:
I feel like we shouldn't be using InitSyncInfo here - I don't think there's any situation we'd want to be creating SyncInfo if for whatever reason it doesn't exist. But see comment above about maybe just updating db.RequireResync based on line 223.

Reply:
Addressed with a refactor, I didn't like this either.